xstream/src/java/com/thoughtworks/xstream/converters/extended/LocalDateTimeConverter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2017 XStream Committers.
+ * All rights reserved.
+ *
+ * The software in this package is published under the terms of the BSD
+ * style license a copy of which has been included with this distribution in
+ * the LICENSE.txt file.
+ *
+ * Created on 13. January 2017 by Matej Cimbora
+ */
+package com.thoughtworks.xstream.converters.extended;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+import com.thoughtworks.xstream.converters.ConversionException;
+import com.thoughtworks.xstream.converters.SingleValueConverter;
+
+/**
+ * Converts a {@link LocalDateTime} to a string.
+ *
+ * @author Matej Cimbora
+ */
+public class LocalDateTimeConverter implements SingleValueConverter {
+
+    private static final DateTimeFormatter FORMATTER;
+
+    static {
+        FORMATTER = new DateTimeFormatterBuilder()
+                .appendPattern("uuuu-MM-dd'T'HH:mm:ss")
+                .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+                .toFormatter();
+    }
+
+    @Override
+    public boolean canConvert(Class<?> type) {
+        return type.equals(LocalDateTime.class);
+    }
+
+    @Override
+    public String toString(Object obj) {
+        if (obj == null) {
+            return null;
+        }
+        LocalDateTime localDateTime = (LocalDateTime) obj;
+
+        return FORMATTER.format(localDateTime);
+    }
+
+    @Override
+    public Object fromString(String str) {
+        try {
+            return LocalDateTime.parse(str);
```

Comment:
I see your point - and it's good point and this isn't a major problem either way - but I 'd argue:
- Read and write should be in sync. If read allows more than write writes, it means that reading an XML file and writing the same content again can change the output. That's a bit of pain for some use cases (think xml solver fragments in optaplanner benchmark's report).
- It should fail fast on non-compliant XSD dates.
- If it doesn't fail fast, it's like browsers not fail fasting on invalid HTML but every browser just rendering it differently. By fail fasting the users would write valid HTML.

That being said, I 'd say:
- Consistently with the other converter inside xstream is most important, @joehni can tell it - and in the end it's his call anyway :)
- I'd argue "when in doubt, leave it out" (Josh Bloch quoth about API design): when users request this smartness, it can always be added later. When we add this smartness now, it can never be removed later (even when almost no one uses it and it prevents it from doing other useful smartness).